### ISSUE TITLE
Update to logic for displaying Board Correspondence 

### ISSUE THREAD

**Issue by shrayshray** [datamade/la-metro-councilmatic#739]
An update to the logic established in #477 for when to display Board Correspondence items - previously, the logic required these 3 fields to be set as follows:
MatterTypeName: Board Correspondence, MatterStatusName: Agenda Ready, MatterRestrictViewViaWeb: False

MatterStatusName should be updated to MatterStatusName: **Sent**

@hancush can this please be prioritized for immediately after reCAPTCHA is complete?

**Comment by hancush** [comment]
Hi, @shrayshray, this is the current logic we use to determine whether to show a bill.

https://github.com/datamade/la-metro-councilmatic/blob/61f7b11e7c046e424af30ec61a77509655350354/lametro/models.py#L47-L78

tl;dr - We don't currently filter on `MatterStatusName`. We show all board correspondence that is not registered as private (`MatterRestrictViewViaWeb` is True, or `MatterStatusName` is "Draft" or "TO BE REMOVED"). 

We don't capture `MatterStatusName` in the scrape, so we don't have that information readily on hand to apply the filter. Before we make changes, is it possible to set `MatterRestrictViewViaWeb` to True for board correspondence you are not ready to show on the board agendas site? Let me know either way and we can go from there!

**Comment by camachoo** [comment]
@hancush 

> We don't currently filter on `MatterStatusName`. We show all board correspondence that is not registered as private (`MatterRestrictViewViaWeb` is True, or `MatterStatusName` is "Draft" or "TO BE REMOVED").

Two things to address here. 1) `MatterStatusName` will not show "TO BE REMOVED".  "TO BE REMOVED" is located in `MatterBodyName`. 2) Maybe I'm looking at this wrong, but isn't checking `MatterStatusName` for "Draft" the same as filtering?

> is it possible to set `MatterRestrictViewViaWeb` to True for board correspondence you are not ready to show on the board agendas site?

Yes, it is.

**Comment by hancush** [comment]
>  "TO BE REMOVED" is located in MatterBodyName.

Whoops, @camachoo, you're totally right. I misread – [we check whether `MatterBodyName` equals "TO BE REMOVED"](https://github.com/opencivicdata/scrapers-us-municipal/blob/84e3250718c8313cd097c68e7cfda9e6e508bf14/lametro/bills.py#L52-L59) as a condition for marking a bill as private.

> isn't checking MatterStatusName for "Draft" the same as filtering?

You can think of it as an indirect filter, sure. If the status is "Draft", the body name is "TO BE REMOVED" or `MatterRestrictViewViaWeb` is True, then we mark the bill as private in our database during the scrape. Then, the app filters out private bills when determining what to show on the board agendas site.

---

If `MatterRestrictViewViaWeb` is a reliable indicator of when to show board correspondence (i.e., it is always `True` if the board correspondence should not be shown), then the existing logic around private bills should handle this case.

**Comment by camachoo** [comment]
@hancush, My hesitation in only looking if `MatterRestrictViewViaWeb` is False is an acceptable solution for posting to the web. Since this file type doesn't follow the normal agenda process, I feel as though there needs to be an additional layer of protection like using `MatterStatusName` as an extra check.  It also makes posting of this type of report intentional. How difficult would it be to add this extra layer of protection?

**Comment by hancush** [comment]
@camachoo Got it. It shouldn't be too tough. Since we aren't currently importing `MatterStatusName`, I'd suggest making the change in the scraper. Namely, I'd add a condition to mark a bill as private if:

- MatterTypeName is "Board Box" or "Board Correspondence" and
- MatterTypeStatus is not "Sent"

Does that seem right to you?

**Comment by shrayshray** [comment]
@hancush that seems right to me!

**Comment by hancush** [comment]
Great, thanks, @shrayshray! I'll draft something in the morning and let you know when it's staged.

**Comment by hancush** [comment]
@camachoo @shrayshray Ok, I've staged this change in scrapers that populate the staging instance. When would you two like to do test data entry?

**Comment by camachoo** [comment]
Let's test Friday morning. I'll create three test files to make sure only the expected one single report shows.

Omar

________________________________

**Comment by shrayshray** [comment]
@camachoo aye, aye!

**Comment by camachoo** [comment]
Last night I setup some files TMP-0101, TMP-0119, and TMP-0121 for this test.

Expected Behavior:
TMP-0101 - Illustrates an accidental marking of `MatterRestrictViewViaWeb` is False.
TMP-0119 - Illustrates an intentional release of a Board Correspondence with a workflow approval that results in `MatterStatusName` is Sent and `MatterRestrictViewViaWeb` is manually set to False.
TMP-0121 - Illustrates an incomplete release of a Board Correspondence with a workflow approval that results in `MatterStatusName` is Sent but `MatterRestrictViewViaWeb` is still True.

**Comment by hancush** [comment]
Looks like we've got 0119: https://lametro-upgrade.datamade.us/board-report/tmp-0119/

But not the others:

- https://lametro-upgrade.datamade.us/board-report/tmp-0101/
- https://lametro-upgrade.datamade.us/board-report/tmp-0121/

🎊

**Comment by hancush** [comment]
The change in scrape logic was deployed.

**Comment by hancush** [comment]
Board correspondence has not yet been added. I'll open a separate issue as a reminder to touch base on this logic once it is added.

**Comment by hancush** [comment]
Actually, I just updated the title here. Moving back into the icebox until data comes in.
